layouts/shortcodes/contribute.html

```diff
@@ -32,4 +59,132 @@
+                        formResultsWrapper.style.background = "#2ECC40";
+                    } else {
+                        // rejected => red
+                        formResultsWrapper.style.background = "#FF4136";
+                    }
+
+                    // return response formatted for result
+                    if (requestStatus) {
+                        return response.json();
+                    } else {
+                        return response.text();
+                    }
+                })
+                .then((result) => {
+                    // display JSON if request valid, otherwise display text
+                    if (requestStatus) {
+                        // display "pull request created" 
+                        formResultsIntroValid.style.display = "inline-block";
+                        
+                        // assign result
+                        formResults.textContent = result.link;
+
+                        // assign href from response result
+                        formResultsLink.href = result.link;
+                        // if <a> not already on formResult, append it
+                        if (!formResultsLink.hasChildNodes()) {
+                            formResultsWrapper.appendChild(formResultsLink);
+                            formResultsLink.appendChild(formResults);
+                        }
+                    } else {
+                        // remove the <a> link if it exists
+                        if (document.querySelector("#formResultsLink")) {
+                            formResultsWrapper.removeChild(document.querySelector("#formResultsLink"));
+
+                            // removing formResultsLink also removed its child,
+                            // so we have to append formResults back
+                            formResultsWrapper.appendChild(formResults);
+                        }
+
+                        // hide "pull request created" 
+                        formResultsIntroValid.style.display = "none";
+
+                        // assign result
+                        formResults.textContent = result;
+                    }
+                })
+                .catch((error) => {
+                    // catch any errors with fetching API
+                    console.log("Error: ", error);
+                });
+            
+            // display form results
+            formResultsWrapper.style.display = "inline-block";
+
+            // clear the form
+            form.reset();
+
+            // submitting should not redirect
+            return false;
+        }
+
+    });
+
+    // https://gist.github.com/UziTech/3b926e732e3a6391b2ab
+    function trimAll(obj) {
+        for (var prop in obj) {
+            if (typeof obj[prop] === "string") {
+                obj[prop] = obj[prop].trim();
+            } else {
+                trimAll(obj[prop]);
+            }
+        }
+    }
+
+    function formToContribute(formData) {
```

Comment:
how about a better name to indicate what the function is doing?
`convertFormToObject`?